### DISCUSSION TITLE
Building on MM4M365 - will there be more?

### DISCUSSION THREAD

**Discussion by ASK-ZIV** [MicrosoftDocs/microsoft-365-community#400] [Q&A]
Recognizing that MM4M365 is focused on the business, I'm curious to know if anyone has plans to build out the capabilities to include platform administration capabilities?

In my day job, I work for a 5000+ employee organization. We have recently undertaken a corporate-wide digital transformation and we are incorporating M365. I am the SharePoint business solution lead representing the business requirements. I personally have a very strong technical background especially in the SharePoint space (since 2001). However, I am not in IT and therefore have difficulty in generating "technical growth".  

Are there plans to create a technical or support capability equivalent? I realize that it might be difficult to keep it platform agnostic especially with SharePoint but as we develop our business, we need to be able implement features. Our IT uses the "if they don't ask for it, why learn about it" approach (reactive with poor results).

I'm working with an organizational change agent for our implementation. We are using MM4M365 to plan our engagement with the business and it's providing valuable structured mapping. I'd like to do the same with SPO admins and site owners. If anyone has done work in this area, I'd love to connect.

I have a specific use case: A technical decision was made to disable the "authoritative news site" setting for our primary corporate communications site (which I asked to be enabled). Corporate communications "did not like the way it looked" on their source page.  The site owner/designer ask me for help because she knew how important that setting is in a modern intranet. In my dev environment. I created a display using the modern CQWP SPFx web part and 50 lines of html to achieve the look they wanted in less than 2 hours. The customer was happy but the technical team rejected it because "SPFx is a bad idea".  This tells me their capability level is not where it needs to be. Suggestions?

**Comment by sympmarc** [comment]
@ASK-ZIV - Welcome! It sounds like you have a set of challenges you're working through and perhaps we don't cover in the existing competency docs - yet. One of the outstanding ones on our roadmap is around Infrastructure. @ToddKlindt is working on an early draft of this, but we haven't published anything yet.

Great to hear the MMM4M365 is proving useful in your planning, though. We'd love to hear more about how you are using it, what works, what doesn't, etc. Are you interested in contributing to the effort?

cc: @eemancini @SimonDoy @SimonJHudson

**Reply by ASK-ZIV** [reply]
Yes I am interested. I'm scheduled for a Sharing is Caring session on the 26th. Once complete, I'm willing to contribute in any way I can.

**Reply by SimonDoy** [reply]
@ASK-ZIV 
Thanks for reaching out, I'd been Interested in unpacking the SPFX is a bad idea and maturity model capability. Do you know if it related to how the team currently build applications? Maybe they currently build applications using .NET / MVC or Java server side code and so are not comfortable with the shift to client side JavaScript approaches?

**Reply by ASK-ZIV** [reply]
@SimonDoy 
It's actually the result of several things. We do use all of the development technologies you mentioned but for other systems/platforms. 

SharePoint in our organization is under-utilized in general. As one of the capability levels mentions, we are using it mostly for file storage and document collaboration. We do have silos of more advanced usage (taxonomies, managed properties, etc) but those groups are typically engineering groups who "figure things out" on their own when they can. By that I mean classic script editor to achieve the visual results they're looking for. 

SharePoint development in our organization has never really been all that mature. The primary reason for this is that the business has never developed the users of SharePoint to the point where they understand the benefits of better IA practices. Site Collection administration for example was always limited to IT support. Hence the "if they don't ask for it..." statement I made earlier. The only "real" SharePoint Web part development that has been done has been limited to a handful of single-use dashboards.  

The initial reaction to SPFx is that IT does not trust anything with the term "open source" embedded in any documentation. Being a part-time developer/integrator outside my day job, I have more experience (sadly) with SPFx than IT. I believe this is only a matter of time for them to understand. 

What I think would help is a road map of sorts to better understand SPFx. Whether that's building components ourselves or hiring someone to do it - our IT security folks need help to properly identify and mitigate risk.

Sorry for the long answer. Quite simply its lack of technical maturity with extending SharePoint.

**Reply by sympmarc** [reply]
@SimonDoy and @SimonJHudson are working on the Applications & Services competency article. Maybe it's time to give it some fresh air for input? @ASK-ZIV, it sounds like you could provide some useful input.

**Comment by sympmarc** [comment]
@ASK-ZIV One other thought on the SPFx approach... it's definitely not a bad idea from a technology approach. 
But that doesn't sound like it's the actual issue in your specific case. Building an SPFx Web Part to display authoritative news in a different way seems like it might possibly be a bad idea. Microsoft has created that functionality and it looks the way it looks.

In the old days, we would have written code (JavaScript or server side code) to make it look different. These days, we try *very* hard to convince people that the way something like this in the UI works is the right choice. It really comes down to where customizations will add the most value for the organization.

I'd argue you could be taking on technical debt if the authoritative news capabilities evolve over time and you can't take advantage of them. Custom solutions ought to be focused on the things that make your organization truly effective and efficient.

But to be very clear: SPFx is the development model you should be using.

**Reply by ASK-ZIV** [reply]
@sympmarc 
I couldn't agree more and I have a complete understanding of what you're saying. Thank you for that.

Our Corporate Communications Department has one of the first production SPO sites. They are moving their legacy communication site along with legacy thinking into the SPO space. Essentially this is a news article repository where users can see the latest news and browse through archived items. This is how our existing model works - users are required to navigate to the one and only corporate news site. 

I know that this is not the modern approach. As I read in countless articles, news needs to surface where the user works whether that's Teams, SharePoint, Outlook, etc. Our Communication group hasn't got there yet. 

They are using a news web part to surface news on their "Corp-x News Site" landing page. When they see the "Corp-x News Site" banner on every one of their articles, it seems cluttered and redundant. Make sense? Remember where they're coming from. 

So the technical debt that you refer would be limited to that one page. Instead, they have turned off authoritative news all together. The header is gone (which they like) but "Corp-x News" will just be "noise" everywhere else because there is no indication of anything else. See my problem?

**Reply by sympmarc** [reply]
I don't envy you your problems, but it's stuff many orgs deal with on their journeys. Rolling up news to the Intranet home page is the step which might make the site link make more sense to them. Couple that with the fact that people _already will see news in other locations_ (search and the SharePoint Home Page, for example) and having that site link makes wonderful sense.

In other words, I'd recommend playing out the eventualities rather than writing code to placate them. You'll end up turning off that code eventually, and the change will be unnecessary.

**Reply by ASK-ZIV** [reply]
Thanks @sympmarc. The approach that I landed on is very similar to your suggestion. Not by design but by understanding that transformation and change management is a two-way exchange. You're right that it's only a matter of time. In my case there is no code to turn off since that wasn't accepted in the first place.  Rather, the authoritative news site setting will be turned on again.

I really appreciate the responses you've provided. It validates and improves my understanding and provides me with some "backing" so to speak when I take these items to our team and to the business. Thanks for making yourself available.

**Reply by sympmarc** [reply]
All that said - and I'm glad it was helpful - we'd love your input on the Maturity Model as you have time.

**Comment by SimonJHudson** [comment]
Hi all
Catching up on the conversation a bit. I'm not qualified to comment too much on the SPFx side (other than supporting Marc's comment about using out of the box whenever possible), however the debate does raise an interesting thought on trying to include something on SPFx in the Development competency we are working on. The reluctance of IT to allow anything with open source components is interesting. Thanks for the input.

Simon Hudson
